### PR TITLE
Use libSwiftScan API to query compilation "Target Info"

### PR DESCRIPTION
**Pull request by artemcm**

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test macOS Platform

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test macOS platform

**Review by nkcsgexi — APPROVED** [review]
Glad we finally got to the bottom of the memory corruption issue👍

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2908,6 +2913,11 @@ extension Driver {
       capturingJSONOutputAs: FrontendTargetInfo.self,
       forceResponseFiles: false,
       recordedInputModificationDates: [:]).target.triple
+//    return try Self.computeTargetInfo(target: nil, targetVariant: nil,
```
Comment (owenv) [inline Sources/SwiftDriver/Driver/Driver.swift:2916]: Should this be commented out?

File: Sources/SwiftDriver/Execution/DriverExecutor.swift [hunk 2]
```diff
@@ -87,6 +87,7 @@ extension DriverExecutor {
                              capturingJSONOutputAs outputType: T.Type,
                              forceResponseFiles: Bool,
                              recordedInputModificationDates: [TypedVirtualPath: TimePoint]) throws -> T {
+    //print(">>> EXEC: \(try self.description(of: job, forceResponseFiles: false))")
```
Comment (owenv) [inline Sources/SwiftDriver/Execution/DriverExecutor.swift:90]: leftover from debugging?

**Review by owenv — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

File: Sources/SwiftDriver/Execution/DriverExecutor.swift (hunk 2 above)
Comment (artemcm) [reply Sources/SwiftDriver/Execution/DriverExecutor.swift]: Yup. Removed. Thanks for catching.

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (artemcm) [reply Sources/SwiftDriver/Driver/Driver.swift]: Also a debugging leftover. Thanks!

**Comment by artemcm** [comment]
@swift-ci test

**Pull request merged**